### PR TITLE
[Form] Fixed ContextErrorException in FileType

### PR DESCRIPTION
**Pull request by chihiro-adachi**
| Q             | A
| ------------- | ---
| Branch?       |  2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | #25063 
| License       | MIT
| Doc PR        | --

Fixed an issue that ContextErrorException occurs when multiple is enabled.

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/FileType.php [hunk 1]
```diff
@@ -34,8 +34,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
             if ($options['multiple']) {
                 $data = array();
+                $files = $event->getData();
 
-                foreach ($event->getData() as $file) {
+                if (!is_array($files)) {
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Extension/Core/Type/FileType.php]: What about checking for `null` here first and then dropping everything afterwards that is not an array?

Comment (xabbuh) [reply src/Symfony/Component/Form/Extension/Core/Type/FileType.php]: or maybe just:

```php
if (!is_array($files)) {
    $files = array(null);
}
```

Comment (chihiro-adachi) [reply src/Symfony/Component/Form/Extension/Core/Type/FileType.php]: @xabbuh Thank you, I fixed it.

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/Form/Extension/Core/Type/FileType.php [hunk 2]
```diff
@@ -34,8 +34,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                    $files = array(null);
```
Comment (issei-m) [inline src/Symfony/Component/Form/Extension/Core/Type/FileType.php]: `$files = array();` also looks more better to me, it reduces the redundant traverse over them and `RequestHandlerInterface::isFileUpload()` method call.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @chihiro-adachi.

File: src/Symfony/Component/Form/Extension/Core/Type/FileType.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Form/Extension/Core/Type/FileType.php]: fixed while merging

**Pull request closed without merging**